dspace/modules/api/src/main/java/org/dspace/JournalUtils.java

```diff
@@ -560,240 +349,43 @@ public static void writeManuscriptToDB(Context context, Manuscript manuscript) t
-            pBean.setEmail(manuscript.correspondingAuthor.email);
+    public static Manuscript getManuscriptFromManuscriptStorage (String manuscriptNumber, DryadJournalConcept journalConcept) {
+        Manuscript result = null;
+        String journalID = journalConcept.getJournalID();
+        if (journalConcept == null) {
+            throw new RuntimeException ("no journalID " + journalID);
         }
-        if (manuscript.optionalProperties != null) {
-            String issn = manuscript.optionalProperties.get("ISSN");
-            if (issn != null) {
-                pBean.setJournalISSN(issn);
-            }
-        }
-        ArrayList<String> subjectKeywords = new ArrayList<String>();
-        for (String keyword : manuscript.keywords) {
-            subjectKeywords.add(keyword);
-        }
-        pBean.setSubjectKeywords(subjectKeywords);
-
-        if (manuscript.isSubmitted()) {
-            pBean.setSkipReviewStep(false);
-        } else if (manuscript.isAccepted() || manuscript.isRejected()) {
-            pBean.setSkipReviewStep(true);
-        }
-        return pBean;
-    }
-
-    public static PublicationBean getPublicationBeanFromManuscriptStorage (String manuscriptNumber, String selectedJournalId) {
-        PublicationBean pBean = null;
-        Context context = null;
         try {
-            context = new Context();
-            Concept journalConcept = JournalUtils.getJournalConceptByShortID(context, selectedJournalId);
             // canonicalize the manuscriptNumber:
-            manuscriptNumber = JournalUtils.getCanonicalManuscriptID(context, manuscriptNumber, selectedJournalId);
+            manuscriptNumber = JournalUtils.getCanonicalManuscriptID(manuscriptNumber, journalID);
             // first, look for a matching manuscript in the database:
-            List<Manuscript> manuscripts = JournalUtils.getManuscriptsMatchingID(selectedJournalId,manuscriptNumber);
+            List<Manuscript> manuscripts = JournalUtils.getManuscriptsMatchingID(journalID, manuscriptNumber);
             if (manuscripts.size() > 0) {
                 log.info("found manuscript " + manuscriptNumber + " in database");
-                pBean = JournalUtils.getPublicationBeanFromManuscript(manuscripts.get(0));
+                result = manuscripts.get(0);
             } else {
-                // if nothing, look in the metadata directory:
-                String journalPath = "";
-                journalPath = JournalUtils.getMetadataDir(journalConcept);
-                pBean = JournalUtils.getDataFromPublisherFile(manuscriptNumber, selectedJournalId, journalPath);
+                // if nothing, use the deprecated metadata files (this can be deleted after we completely move away from them):
+                File file = new File(journalConcept.getMetadataDir() + File.separator + manuscriptNumber + ".xml");
+                try {
+                    result = ManuscriptToLegacyXMLConverter.convertInternalXMLToManuscript(file);
```

Comment:
convertInternalXMLToManuscript above does not throw an exception (returns null instead)